File: RadixSort/RadixSort.js
function random(min, max){
    return Math.floor(Math.random()*(max-min+1))+min;
}

function printarray(a){
    for(let i=0;i<a.length;i++){
    console.log(a[i])
}
}

function countingSortForRadix(inputArr, place) {
    let outputArr = new Array(inputArr.length).fill(0);
    let countArr = new Array(10).fill(0);
    for (let i = 0; i < inputArr.length; i++) {
        let digit = Math.floor((inputArr[i] / place) % 10);
        countArr[digit]++;
    }
    for (let i = 1; i < countArr.length; i++) {
        countArr[i] += countArr[i - 1];
    }
    for (let i = inputArr.length - 1; i >= 0; i--) {
        let digit = Math.floor((inputArr[i] / place) % 10);
        outputArr[countArr[digit] - 1] = inputArr[i];
        countArr[digit]--;
    }
    for (let i = 0; i < inputArr.length; i++) {
        inputArr[i] = outputArr[i];
    }
}

function radix_sort(inputArr) {
    let maxVal = Math.max(...inputArr);
    for (let place = 1; Math.floor(maxVal / place) > 0; place *= 10) {
        countingSortForRadix(inputArr, place);
    }
}
let arr=new Array(15)

for(let i=0;i<arr.length;i++){
    arr[i]=random(1,100);
}
console.log("\nArreglo Original: ");

printarray(arr)
radix_sort(arr)
console.log("\nArray Ordenado: ")
printarray(arr)

File: RadixSort/RadixSort.cs
using System;
using System.Collections.Generic;

class Program
{
    static void randomnumber(int[] a)
    {
        Random rand = new Random();
        for (int i = 0; i < a.Length; i++)
        {
            a[i] = rand.Next(1, 100);
        }
    }

    static void printarray(int[] a)
    {
        for (int i = 0; i < a.Length; i++)
        {
            Console.Write(a[i] + " ");
        }
        Console.WriteLine();
    }
    static void countingSort(int[] arr, int exp)
    {
        int n = arr.Length;
        int[] output = new int[n];
        int[] count = new int[10];

        for (int i = 0; i < n; i++)
            count[(arr[i] / exp) % 10]++;

        for (int i = 1; i < 10; i++)
            count[i] += count[i - 1];

        for (int i = n - 1; i >= 0; i--)
        {
            output[count[(arr[i] / exp) % 10] - 1] = arr[i];
            count[(arr[i] / exp) % 10]--;
        }

        for (int i = 0; i < n; i++)
            arr[i] = output[i];
    }

    static void radix_sort(int[] arr)
    {
        int max = arr[0];
        for (int i = 1; i < arr.Length; i++)
            if (arr[i] > max)
                max = arr[i];

        for (int exp = 1; max / exp > 0; exp *= 10)
            countingSort(arr, exp);
    }
    static void Main()
    {
        int[] arr = new int[15];
        randomnumber(arr);
        Console.WriteLine("Array Original: ");
        printarray(arr);
        radix_sort(arr);
        Console.WriteLine("Array Ordenado con Radix Sort: "); 
        printarray(arr);
    }
}